syntax = "proto3";
import "auth.proto";
import "get_access_token.proto";
import "get_user.proto";
import "update_user.proto";
import "delete_user.proto";

service IdentityService {
  rpc logout(LogoutRequest) returns (BaseResponse){}
  rpc auth(AccessToken) returns (AuthResponse){}
  rpc login(LoginRequest) returns (CredentialsResponse){}
  rpc register(RegisterRequest) returns (CredentialsResponse){}
  rpc get_new_access_token(GetNewAccessTokenRequest) returns (GetNewAccessTokenResponse){}
  rpc get_user_by_id(UserByIdRequest) returns (UserByIdResponse){}
  rpc get_users_by_id(UsersByIdRequest) returns (UsersByIdResponse){}
  rpc update_user(UpdateUserRequest) returns (CredentialsResponse){}
  rpc delete_user(DeleteUserRequest) returns (BaseResponse){}
}

message BaseResponse {
  int64 status_code = 1;
  optional string message = 2;
}
